libs/ledger/src/lib/ledger-export-form.tsx

```diff
@@ -0,0 +1,173 @@
+import { toNanoSeconds } from '@vegaprotocol/utils';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useEnvironment } from '@vegaprotocol/environment';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+  Button,
+  Loader,
+  TradingFormGroup,
+  TradingInput,
+} from '@vegaprotocol/ui-toolkit';
+import { t } from '@vegaprotocol/i18n';
+import { useAssetsDataProvider } from '@vegaprotocol/assets';
+import { localLoggerFactory } from '@vegaprotocol/logger';
+import { formatForInput } from '@vegaprotocol/utils';
+import { subDays } from 'date-fns';
+
+const DEFAULT_EXPORT_FILE_NAME = 'ledger_entries.csv';
+
+const getProtoHost = (vegaurl: string) => {
+  const loc = new URL(vegaurl);
+  return `${loc.protocol}//${loc.host}`;
+};
+
+export const LedgerExportForm = ({ partyId }: { partyId: string }) => {
+  const now = useRef(new Date());
+  const defaultFrom = subDays(now.current, 7);
+  const [dateFrom, setDateFrom] = useState(formatForInput(defaultFrom));
+  const [dateTo, setDateTo] = useState('');
+  const maxFromDate = formatForInput(new Date(dateTo || now.current));
+  const maxToDate = formatForInput(now.current);
+  const { data } = useAssetsDataProvider();
```

Comment:
Pass in assets from ledger-container in app

Reply:
done